### PR TITLE
Bump up ibc-go to v7.0.0

### PR DESCRIPTION
**Pull request by siburu**
## Summary
- Dependencies
    - ibc-go: v4.0.0 => v7.0.0
    - tendermint => cometbft
- Protobuf management
    - use `buf` instead of the `third_party` directory
- renew simapp based on ibc-go@v7/testings/simapp
- fix `core.Header` interface
- remove old unused functions and variables

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,114 +1,168 @@
 module github.com/hyperledger-labs/yui-relayer
 
-go 1.18
+go 1.19
```
Comment (bluele) [inline go.mod:3]: Looks like you use go1.20 in the test. Should we use the same version in the repo?

**Review by bluele — COMMENTED** [review]
Also, you might want to update the CONTRACT here: https://github.com/hyperledger-labs/yui-relayer/pull/87/files#diff-3ebe73b99a006b3bde594abcedd2812ffc28cbdf68ad4fbec70e5be4c6bdc1e8R367

File: go.mod (hunk 1 above)
Comment (siburu) [reply go.mod]: Thank you. I've bumped up the versions.

**Review by bluele — APPROVED** [review]
LGTM, Great work! @siburu

**Pull request merged**